core/include/scipp/core/time_point.h

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+/// @file
+#pragma once
+
+#include <stdint.h>
+
+namespace scipp::core {
+
+/// Time-point similar to std::chrono::time point but without compile-time unit.
+class time_point {
+public:
+  constexpr time_point() noexcept : m_duration(0){};
```

Comment:
```suggestion
  constexpr time_point() noexcept = default;
```
should be enough, since we have initialized `m_duration` below.